File: AssetManager/Data/DataClasses/DataClassMapping/MappableClasses.vb
Public Class DeviceObject
    Inherits DataMapping

    Sub New()

    End Sub

    Sub New(data As Object)
        Me.MapClassProperties(Me, data)
    End Sub

    <DataColumnName(DevicesCols.AssetTag)>
    Public Property AssetTag As String

    <DataColumnName(DevicesCols.Description)>
    Public Property Description As String

    <DataColumnName(DevicesCols.EQType)>
    Public Property EquipmentType As String

    <DataColumnName(DevicesCols.Serial)>
    Public Property Serial As String

    <DataColumnName(DevicesCols.Location)>
    Public Property Location As String

    <DataColumnName(DevicesCols.CurrentUser)>
    Public Property CurrentUser As String

    <DataColumnName(DevicesCols.MunisEmpNum)>
    Public Property CurrentUserEmpNum As String

    <DataColumnName(DevicesCols.PurchaseDate)>
    Public Property PurchaseDate As Date

    <DataColumnName(DevicesCols.ReplacementYear)>
    Public Property ReplaceYear As String

    <DataColumnName(DevicesCols.OSVersion)>
    Public Property OSVersion As String

    <DataColumnName(DevicesCols.PhoneNumber)>
    Public Property PhoneNumber As String

    <DataColumnName(DevicesCols.DeviceUID)>
    Public Property GUID As String

    <DataColumnName(DevicesCols.PO)>
    Public Property PO As String

    <DataColumnName(DevicesCols.Status)>
    Public Property Status As String

    <DataColumnName(DevicesCols.Trackable)>
    Public Property IsTrackable As Boolean

    <DataColumnName(DevicesCols.SibiLinkUID)>
    Public Property SibiLink As String

    <DataColumnName(DevicesCols.HostName)>
    Public Property HostName As String


    Public Property FiscalYear As String
    Public Property Note As String

    Public Property Tracking As New DeviceTrackingObject
    Public Property Historical As New DeviceHistoricalObject

End Class

Public Class RequestObject
    Inherits DataMapping

    Sub New()
        Me.GUID = System.Guid.NewGuid.ToString
    End Sub

    Sub New(data As Object)
        Me.MapClassProperties(Me, data)
    End Sub


    <DataColumnName(SibiRequestCols.UID)>
    Public Property GUID As String
    <DataColumnName(SibiRequestCols.RequestUser)>
    Public Property RequestUser As String
    <DataColumnName(SibiRequestCols.Description)>
    Public Property Description As String
    <DataColumnName(SibiRequestCols.DateStamp)>
    Public Property DateStamp As Date
    <DataColumnName(SibiRequestCols.NeedBy)>
    Public Property NeedByDate As Date
    <DataColumnName(SibiRequestCols.Status)>
    Public Property Status As String
    <DataColumnName(SibiRequestCols.Type)>
    Public Property RequestType As String
    <DataColumnName(SibiRequestCols.PO)>
    Public Property PO As String
    <DataColumnName(SibiRequestCols.RequisitionNumber)>
    Public Property RequisitionNumber As String
    <DataColumnName(SibiRequestCols.ReplaceAsset)>
    Public Property ReplaceAsset As String
    <DataColumnName(SibiRequestCols.ReplaceSerial)>
    Public Property ReplaceSerial As String
    <DataColumnName(SibiRequestCols.RequestNumber)>
    Public Property RequestNumber As String
    <DataColumnName(SibiRequestCols.RTNumber)>
    Public Property RTNumber As String


    Public RequestItems As DataTable
End Class

Public Class DeviceHistoricalObject
    Inherits DataMapping

    Sub New()

    End Sub

    Sub New(data As Object)
        Me.MapClassProperties(Me, data)
    End Sub

    <DataColumnName(HistoricalDevicesCols.ChangeType)>
    Public Property ChangeType As String
    <DataColumnName(HistoricalDevicesCols.HistoryEntryUID)>
    Public Property GUID As String
    <DataColumnName(HistoricalDevicesCols.Notes)>
    Public Property Note As String
    <DataColumnName(HistoricalDevicesCols.ActionUser)>
    Public Property ActionUser As String
    <DataColumnName(HistoricalDevicesCols.ActionDateTime)>
    Public Property ActionDateTime As Date
End Class

Public Class DeviceTrackingObject
    Inherits DataMapping

    Sub New()

    End Sub

    Sub New(data As DataTable)
        Me.MapClassProperties(Me, data)
    End Sub

    <DataColumnName(TrackablesCols.CheckoutTime)>
    Public Property CheckoutTime As Date
    <DataColumnName(TrackablesCols.DueBackDate)>
    Public Property DueBackTime As Date
    <DataColumnName(TrackablesCols.CheckinTime)>
    Public Property CheckinTime As Date
    <DataColumnName(TrackablesCols.CheckoutUser)>
    Public Property CheckoutUser As String
    <DataColumnName(TrackablesCols.CheckinUser)>
    Public Property CheckinUser As String
    <DataColumnName(TrackablesCols.UseLocation)>
    Public Property UseLocation As String
    <DataColumnName(TrackablesCols.Notes)>
    Public Property UseReason As String
    <DataColumnName(DevicesCols.CheckedOut)>
    Public Property IsCheckedOut As Boolean
    <DataColumnName(TrackablesCols.Notes)>
    Public Property CheckinNotes As String
    <DataColumnName(TrackablesCols.DeviceUID)>
    Public Property DeviceGUID As String
End Class

File: AssetManager/Data/DataClasses/DataClassMapping/MappableClasses.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Threading.Tasks;
namespace AssetManager
{
    public class DeviceObject : DataMapping
    {


        public DeviceObject()
        {
        }

        public DeviceObject(object data)
        {
            this.MapClassProperties(this, data);
        }

        [DataColumnName(DevicesCols.AssetTag)]
        public string AssetTag { get; set; }

        [DataColumnName(DevicesCols.Description)]
        public string Description { get; set; }

        [DataColumnName(DevicesCols.EQType)]
        public string EquipmentType { get; set; }

        [DataColumnName(DevicesCols.Serial)]
        public string Serial { get; set; }

        [DataColumnName(DevicesCols.Location)]
        public string Location { get; set; }

        [DataColumnName(DevicesCols.CurrentUser)]
        public string CurrentUser { get; set; }

        [DataColumnName(DevicesCols.MunisEmpNum)]
        public string CurrentUserEmpNum { get; set; }

        [DataColumnName(DevicesCols.PurchaseDate)]
        public System.DateTime PurchaseDate { get; set; }

        [DataColumnName(DevicesCols.ReplacementYear)]
        public string ReplaceYear { get; set; }

        [DataColumnName(DevicesCols.OSVersion)]
        public string OSVersion { get; set; }

        [DataColumnName(DevicesCols.PhoneNumber)]
        public string PhoneNumber { get; set; }

        [DataColumnName(DevicesCols.DeviceUID)]
        public string GUID { get; set; }

        [DataColumnName(DevicesCols.PO)]
        public string PO { get; set; }

        [DataColumnName(DevicesCols.Status)]
        public string Status { get; set; }

        [DataColumnName(DevicesCols.Trackable)]
        public bool IsTrackable { get; set; }

        [DataColumnName(DevicesCols.SibiLinkUID)]
        public string SibiLink { get; set; }

        [DataColumnName(DevicesCols.HostName)]
        public string HostName { get; set; }

        public string FiscalYear { get; set; }
        public string Note { get; set; }

        public DeviceTrackingObject Tracking { get; set; } = new DeviceTrackingObject();
        public DeviceHistoricalObject Historical { get; set; } = new DeviceHistoricalObject();

    }
}
namespace AssetManager
{

    public class RequestObject : DataMapping
    {

        public RequestObject()
        {
            this.GUID = System.Guid.NewGuid().ToString();
        }

        public RequestObject(object data)
        {
            this.MapClassProperties(this, data);
        }

        [DataColumnName(SibiRequestCols.UID)]
        public string GUID { get; set; }

        [DataColumnName(SibiRequestCols.RequestUser)]
        public string RequestUser { get; set; }

        [DataColumnName(SibiRequestCols.Description)]
        public string Description { get; set; }

        [DataColumnName(SibiRequestCols.DateStamp)]
        public System.DateTime DateStamp { get; set; }

        [DataColumnName(SibiRequestCols.NeedBy)]
        public System.DateTime NeedByDate { get; set; }

        [DataColumnName(SibiRequestCols.Status)]
        public string Status { get; set; }

        [DataColumnName(SibiRequestCols.Type)]
        public string RequestType { get; set; }

        [DataColumnName(SibiRequestCols.PO)]
        public string PO { get; set; }

        [DataColumnName(SibiRequestCols.RequisitionNumber)]
        public string RequisitionNumber { get; set; }

        [DataColumnName(SibiRequestCols.ReplaceAsset)]
        public string ReplaceAsset { get; set; }

        [DataColumnName(SibiRequestCols.ReplaceSerial)]
        public string ReplaceSerial { get; set; }

        [DataColumnName(SibiRequestCols.RequestNumber)]
        public string RequestNumber { get; set; }

        [DataColumnName(SibiRequestCols.RTNumber)]
        public string RTNumber { get; set; }

        public DataTable RequestItems;
    }
}
namespace AssetManager
{

    public class DeviceHistoricalObject : DataMapping
    {


        public DeviceHistoricalObject()
        {
        }

        public DeviceHistoricalObject(object data)
        {
            this.MapClassProperties(this, data);
        }

        [DataColumnName(HistoricalDevicesCols.ChangeType)]
        public string ChangeType { get; set; }

        [DataColumnName(HistoricalDevicesCols.HistoryEntryUID)]
        public string GUID { get; set; }

        [DataColumnName(HistoricalDevicesCols.Notes)]
        public string Note { get; set; }

        [DataColumnName(HistoricalDevicesCols.ActionUser)]
        public string ActionUser { get; set; }

        [DataColumnName(HistoricalDevicesCols.ActionDateTime)]
        public System.DateTime ActionDateTime { get; set; }

    }
}
namespace AssetManager
{

    public class DeviceTrackingObject : DataMapping
    {


        public DeviceTrackingObject()
        {
        }

        public DeviceTrackingObject(DataTable data)
        {
            this.MapClassProperties(this, data);
        }

        [DataColumnName(TrackablesCols.CheckoutTime)]
        public System.DateTime CheckoutTime { get; set; }

        [DataColumnName(TrackablesCols.DueBackDate)]
        public System.DateTime DueBackTime { get; set; }

        [DataColumnName(TrackablesCols.CheckinTime)]
        public System.DateTime CheckinTime { get; set; }

        [DataColumnName(TrackablesCols.CheckoutUser)]
        public string CheckoutUser { get; set; }

        [DataColumnName(TrackablesCols.CheckinUser)]
        public string CheckinUser { get; set; }

        [DataColumnName(TrackablesCols.UseLocation)]
        public string UseLocation { get; set; }

        [DataColumnName(TrackablesCols.Notes)]
        public string UseReason { get; set; }

        [DataColumnName(DevicesCols.CheckedOut)]
        public bool IsCheckedOut { get; set; }

        [DataColumnName(TrackablesCols.Notes)]
        public string CheckinNotes { get; set; }

        [DataColumnName(TrackablesCols.DeviceUID)]
        public string DeviceGUID { get; set; }

    }
}
namespace AssetManager
{

    public class AccessGroupObject : DataMapping
    {


        public AccessGroupObject()
        {
        }

        public AccessGroupObject(object data)
        {
            this.MapClassProperties(this, data);
        }

        [DataColumnName(SecurityCols.SecModule)]
        public string AccessModule { get; set; }

        [DataColumnName(SecurityCols.AccessLevel)]
        public int Level { get; set; }

        [DataColumnName(SecurityCols.Description)]
        public string Description { get; set; }

        [DataColumnName(SecurityCols.AvailOffline)]
        public bool AvailableOffline { get; set; }

    }
}
